python/ray/data/datasource/lance_datasource.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from typing import List, Optional
+
+import lance
+from lance import LanceFragment
+
+from ray.data import ReadTask
+from ray.data.block import Block, BlockMetadata
+from ray.data.datasource import Datasource
+from ray.util.annotations import DeveloperAPI
+
+import pyarrow as pa
+
+logger = logging.getLogger(__name__)
+
+
+@DeveloperAPI
+class LanceDatasource(Datasource):
+    """Lance Datasource
+    Read a Lance table as a Ray Dataset
```

Comment:
let's be consistent w/ naming below, `Lance table` -> `Lance dataset`.

Reply:
Updated